### PR TITLE
rest: workflow diff

### PR DESCRIPTION
**Pull request by dinosk**
* Adds REST API endpoint which calls
  reana-workflow-controller to retrieve
  a list of differences between two workflows.

Connects https://github.com/reanahub/reana-client/issues/193

Signed-off-by: Dinos Kousidis <dinos.kousidis@cern.ch>

### CODE REVIEW

File: reana_server/rest/workflows.py [hunk 1]
```diff
@@ -1119,3 +1119,123 @@ def get_workflow_parameters(workflow_id_or_name):  # noqa
...
+        - name: brief
+          in: query
+          description: Optional flag. If set, file contents are examined.
+          required: false
+          type: boolean
+        - name: context_lines
+          in: query
+          description: Optional parameter. Sets number of context lines
+                       for workspace diff output.
+          required: false
```
Comment (diegodelemos) [inline reana_server/rest/workflows.py]: Also default value here.

File: reana_server/rest/workflows.py [hunk 2]
```diff
@@ -1119,3 +1119,123 @@ def get_workflow_parameters(workflow_id_or_name):  # noqa
...
+          type: string
```
Comment (diegodelemos) [inline reana_server/rest/workflows.py]: Type is integer I guess?

File: reana_server/rest/workflows.py [hunk 3]
```diff
@@ -1119,3 +1119,123 @@ def get_workflow_parameters(workflow_id_or_name):  # noqa
...
+        - name: workflow_id_or_name_b
+          in: path
+          description: Required. Analysis UUID or name of the second workflow.
+          required: true
+          type: string
+        - name: brief
+          in: query
+          description: Optional flag. If set, file contents are examined.
+          required: false
+          type: boolean
```
Comment (diegodelemos) [inline reana_server/rest/workflows.py]: We should document here the default value.

File: reana_server/rest/workflows.py (hunk 2 above)
Comment (dinosk) [reply reana_server/rest/workflows.py]: Indeed this is what I tried first, and then after the RC->RS->RWC the type had changed to `string` so I changed it to string. I am not 100% sure, but I believe this happened with the type set to `integer` in both specs.

**Comment by diegodelemos** [comment]
Needs rebase.

**Pull request merged**